packages/teams-js/src/public/barCode.ts

```diff
@@ -0,0 +1,44 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { validateScanBarCodeInput } from '../internal/mediaUtil';
+import { FrameContexts } from './constants';
+import { DevicePermission, ErrorCode } from './interfaces';
+import { runtime } from './runtime';
+
+export namespace barCode {
+  export interface BarCodeConfig {
+    timeOutIntervalInSec?: number;
+  }
+
+  export function scanBarCode(barCodeConfig: BarCodeConfig): Promise<string> {
+    return new Promise<string>(resolve => {
+      ensureInitialized(FrameContexts.content, FrameContexts.task);
```

Comment:
we should check isSupported here

Reply:
Should we add ensureInitialized and isSupported to other two fucntions - hasPermission, and requestPermission - as well?

Reply:
In our convo, we decided yes, we need to add both